File: python/Encapsulation/src/personne.py
class Personne:
    def __init__(self, nom: str, age: int):
        self.nom = nom
        self.age = age

    @property
    def nom(self):
        return self._nom

    @nom.setter
    def nom(self, value):
        if not value or not value.strip():
            raise ValueError("Le nom ne peut pas être vide.")
        self._nom = value.strip()

    @property
    def age(self):
        return self._age

    @age.setter
    def age(self, value):
        if not (0 <= value <= 100):
            raise ValueError("L'âge doit être compris entre 0 et 100.")
        self._age = value

    def __str__(self):
        return f"Personne: {self._nom}, {self._age} ans"

File: java/src/main/java/encapsulation/Personne.java
package encapsulation;

public class Personne {
    private String nom;
    private int age;

    public Personne(String nom, int age) {
        setNom(nom);
        setAge(age);
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        if (nom == null || nom.trim().isEmpty()) {
            throw new IllegalArgumentException("Le nom ne peut pas être vide.");
        }
        this.nom = nom.trim();
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        if (age < 0 || age > 100) {
            throw new IllegalArgumentException("L'âge doit être compris entre 0 et 100.");
        }
        this.age = age;
    }

    @Override
    public String toString() {
        return "Personne: " + nom + ", " + age + " ans";
    }
}